### PR TITLE
Loading Animation Data

### PR DESCRIPTION
**Pull request by Whixx**
Loading the skeleton and animation data from an .fbx file should be working.

### CODE REVIEW

**Review by OtakuFreak13 — DISMISSED** [review]
Test scene runs and animation and bone data seems to load in correctly from what I can see stepping through the "among us" model, model loading function.

File: StortSpelProjekt/Engine/src/Misc/AssetLoader.cpp [hunk 1]
```diff
@@ -545,4 +581,94 @@ Font* AssetLoader::loadFont(LPCWSTR filename, int windowWidth, int windowHeight)
...
+	// Store the scale values
+	for (unsigned int i = 0; i < assimpNodeAnimation->mNumScalingKeys; i++)
+	{
+		nodeAnimation->scalings.push_back(
+			DirectX::XMFLOAT3(
+				assimpNodeAnimation->mScalingKeys[i].mValue.x,
+				assimpNodeAnimation->mScalingKeys[i].mValue.y,
+				assimpNodeAnimation->mScalingKeys[i].mValue.z));
+	}
+}
```
Comment (fredrix96) [inline StortSpelProjekt/Engine/src/Misc/AssetLoader.cpp:647]: Do you need three loops of the same size? Can you instead do one loop with three commands?

File: StortSpelProjekt/Sandbox/src/main.cpp [hunk 2]
```diff
@@ -30,6 +31,7 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR, int nCmdShow)
     //sceneManager->SetSceneToDraw(TimScene(sceneManager));
     //sceneManager->SetSceneToDraw(JockesTestScene(sceneManager));
     //sceneManager->SetSceneToDraw(FredriksTestScene(sceneManager));
+    sceneManager->SetSceneToDraw(WilliamsTestScene(sceneManager));
```
Comment (fredrix96) [inline StortSpelProjekt/Sandbox/src/main.cpp]: make the test inactive before pushing it to develop

File: StortSpelProjekt/Engine/src/Misc/AssetLoader.cpp [hunk 3]
```diff
@@ -545,4 +581,94 @@ Font* AssetLoader::loadFont(LPCWSTR filename, int windowWidth, int windowHeight)
...
+		nodeAnimation->rotationQuaternions.push_back(
+			DirectX::XMFLOAT4(
+				assimpNodeAnimation->mRotationKeys[i].mValue.x,
+				assimpNodeAnimation->mRotationKeys[i].mValue.y,
+				assimpNodeAnimation->mRotationKeys[i].mValue.z,
+				assimpNodeAnimation->mRotationKeys[i].mValue.w));
+	}
+
+	// Store the scale values
+	for (unsigned int i = 0; i < assimpNodeAnimation->mNumScalingKeys; i++)
```
Comment (fredrix96) [inline StortSpelProjekt/Engine/src/Misc/AssetLoader.cpp:639]: Do you need three loops of the same size? instead, can you do one loop with three commands?

**Review by fredrix96 — CHANGES_REQUESTED** [review]

**Review by OtakuFreak13 — APPROVED** [review]
Run after merges so I approve.

**Review by fredrix96 — APPROVED** [review]
A most excellent job. Well done William. Animations seem like a tough task

**Pull request merged**